### DISCUSSION TITLE
LuaSnip Configuration

### DISCUSSION THREAD

**Discussion by WilliamHangXu** [LazyVim/LazyVim#2253] [Q&A] [answered]
Hi all, I'm trying to use LuaSnip with LazyVim. 

1. Which file should I modify so that I can change the trigger key of snippets to <Tab>?
2. I have a collection of .lua files containing my snippets. Where should I put them? I initially put them in a folder in the /nvim directory, but LuaSnip didn't seem to recognize them.

Thank you!

**Comment by dpetka2001** [comment, marked as answer]
1. I don't believe you should do that. Those come from `friendly-snippets` and you would have to change the source files from the plugin. That has the possibility that in a future update of `friendly-snippets` it will revert back to the previous change or you may also not be able to update `friendly-snippets` any more because you will have made local changes to the git state and it will diverge from the remote branch.
You should instead copy that snippet into your custom personal ones and use it with a different prefix.
2. I have my custom snippets in `~/.config/nvim/lua/snippets` and make Luasnip aware of them with the following 
```lua
-- ~/.config/nvim/lua/plugins/luasnip.lua
return {
  {
    "L3MON4D3/LuaSnip"
    dependencies = {
      "rafamadriz/friendly-snippets",
      config = function()
        require("luasnip.loaders.from_vscode").lazy_load()
        require("luasnip.loaders.from_lua").lazy_load({ paths = "./lua/snippets" })
      end,
    },
  },
}
```

**Reply by WilliamHangXu** [reply]
Thank you!

**Reply by weberam2** [reply]
Thank you!

**Reply by LaoYang1994** [reply]
@dpetka2001 I don't know why my setting doesn't work. Could you help me?
```python
return {
  {
    "L3MON4D3/LuaSnip",
    dependencies = {
      {
        "rafamadriz/friendly-snippets",
        config = function()
          require("luasnip.loaders.from_vscode").lazy_load()
          require("luasnip.loaders.from_vscode").lazy_load({ paths = "~/.config/nvim/my-snippets" })
        end,
      },
    },
  },
}

**Reply by dpetka2001** [reply]
No idea. I don't use snippets that much. I had just tested if it worked in the past. But i tried with `lua` snippets whereas you seem to be trying with `vscode` style snippets. Do you have a `package.json` in your directory with the snippets (i believe it's needed for `vscode` style custom snippets) ? I'm afraid you would have to read Luasnip's docs for further help.

PS: also i take it you have enabled the Luasnip Extra right?

**Reply by LaoYang1994** [reply]
@dpetka2001 It's my fault. I missed the `package.json` file. Thanks!
